packages/api/test/user-payment.spec.js

```diff
@@ -154,11 +155,46 @@ describe('userPaymentPut', () => {
       customers.mockCustomers.map(c => c.id).includes(billing.paymentMethodSaves[0].customerId),
       'billing.paymentMethodSaves[0].customerId is in customers.mockCustomers')
   })
+  it('throws an error if no agreement', async function () {
+    const desiredPaymentMethodId = `pm_${randomString()}`
+    const paymentSettings = { paymentMethod: { id: desiredPaymentMethodId }, subscription: { storage: null } }
+    const authorization = createBearerAuthorization(AuthorizationTestContext.use(this).createUserToken())
+    const request = createMockAuthenticatedRequest(createSaveUserPaymentSettingsRequest({ authorization, body: JSON.stringify(paymentSettings) }))
+    const billing = createMockBillingService()
+    const customers = createMockCustomerService()
+    const env = {
+      ...createMockBillingContext(),
+      billing,
+      customers
+    }
+    assert.rejects(
+      async () => await userPaymentPut(request, env),
+      /Missing Terms of Service/
+    )
+  })
+  it('throws an error if bad agreement', async function () {
+    const desiredPaymentMethodId = `pm_${randomString()}`
+    const paymentSettings = { paymentMethod: { id: desiredPaymentMethodId }, subscription: { storage: null }, agreement: 'bad-agreement' }
+    const authorization = createBearerAuthorization(AuthorizationTestContext.use(this).createUserToken())
+    const request = createMockAuthenticatedRequest(createSaveUserPaymentSettingsRequest({ authorization, body: JSON.stringify(paymentSettings) }))
+    const billing = createMockBillingService()
+    const customers = createMockCustomerService()
+    const env = {
+      ...createMockBillingContext(),
+      billing,
+      customers
+    }
+    assert.rejects(
+      async () => await userPaymentPut(request, env),
+      /Invalid Terms of Service/
```

Comment:
wdyt of this mentioning 'agreement' instead of 'Terms of Service' since that's what the field in the paymentSettings object calls it? e.g. it could be an error message like `Unable to save Payment Settings due to unknown agreement value "{paymentSettings.agreement}"`

Also is there a way for these tests we can ensure that the error being thrown will lead to an HTTP 400 response status code? That way even if the frontend doesn't know about the exact underlying error, it can at least use the http response status conventions to try to decide what to do or how to format a generic error for the end-user.